src/OrchardCore.Modules/OrchardCore.Tenants/Workflows/Activities/SetupTenantTask.cs

```diff
@@ -0,0 +1,163 @@
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> AdminPassword
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> DatabaseProvider
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> DatabaseConnectionString
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> DatabaseTablePrefix
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public WorkflowExpression<string> RecipeName
+        {
+            get => GetProperty(() => new WorkflowExpression<string>());
+            set => SetProperty(value);
+        }
+
+        public override IEnumerable<Outcome> GetPossibleOutcomes(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            return Outcomes(T["Done"], T["Failed"]);
+        }
+
+        public override async Task<ActivityExecutionResult> ExecuteAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+        {
+            var tenantNameTask = _expressionEvaluator.EvaluateAsync(TenantName, workflowContext);
+            var siteNameTask = _expressionEvaluator.EvaluateAsync(SiteName, workflowContext);
+            var adminUsernameTask = _expressionEvaluator.EvaluateAsync(AdminUsername, workflowContext);
+            var adminEmailTask = _expressionEvaluator.EvaluateAsync(AdminEmail, workflowContext);
+            var adminPasswordTask = _expressionEvaluator.EvaluateAsync(AdminPassword, workflowContext);
+            var databaseProviderTask = _expressionEvaluator.EvaluateAsync(DatabaseProvider, workflowContext);
+            var databaseConnectionStringTask = _expressionEvaluator.EvaluateAsync(DatabaseConnectionString, workflowContext);
+            var databaseTablePrefixTask = _expressionEvaluator.EvaluateAsync(DatabaseTablePrefix, workflowContext);
+            var recipeNameTask = _expressionEvaluator.EvaluateAsync(RecipeName, workflowContext);
+
+            await Task.WhenAll(tenantNameTask, siteNameTask, adminUsernameTask, adminEmailTask, adminPasswordTask, databaseProviderTask, databaseConnectionStringTask, databaseTablePrefixTask, recipeNameTask);
+
+            if (!ShellHost.TryGetSettings(tenantNameTask.Result?.Trim(), out var shellSettings))
+            {
+                if (!string.IsNullOrWhiteSpace(tenantNameTask.Result))
+                {
+                    shellSettings = new ShellSettings
+                    {
```

Comment:
maybe here we need

                // Creates a default shell settings based on the configuration.
                var shellSettings = _shellSettingsManager.CreateDefaultSettings();